private_membership/rlwe/batch/cpp/parameters.cc

```diff
@@ -0,0 +1,122 @@
+      parameters.log_decomposition_modulus() > kMaxLogDecompositionModulus) {
+    return absl::InvalidArgumentError(
+        "log_decomposition_modulus is out of the valid range");
+  }
+  if (parameters.log_degree() <= 0 || parameters.log_degree() > kMaxLogDegree) {
+    return absl::InvalidArgumentError("log_degree is out of the valid range");
+  }
+  if (parameters.log_t() <= 0 || parameters.log_t() > kMaxLogT) {
+    return absl::InvalidArgumentError("log_t is out of the valid range");
+  }
+  if (parameters.request_modulus_size() <= 0 ||
+      parameters.request_modulus_size() > kMaxRequestModulusSize) {
+    return absl::InvalidArgumentError(
+        "Invalid number of request_modulus entries");
+  }
+  if (parameters.response_modulus_size() <= 0 ||
+      parameters.response_modulus_size() > kMaxResponseModulusSize) {
+    return absl::InvalidArgumentError(
+        "Invalid number of response_modulus entries");
+  }
+  if (parameters.variance() <= 0 || parameters.variance() > kMaxVariance) {
+    return absl::InvalidArgumentError("variance is out of the valid range");
+  }
+  return absl::OkStatus();
+}
+
+absl::Status ValidateShardParameters(
+    const Parameters::ShardParameters& parameters) {
+  if (parameters.number_of_shards() <= 0 ||
+      parameters.number_of_shards() > kMaxNumShards) {
+    return absl::InvalidArgumentError(
+        "number_of_shards is out of the valid range");
+  }
+  if (parameters.number_of_buckets_per_shard() <= 0 ||
+      parameters.number_of_buckets_per_shard() > kMaxNumBucketsPerShard) {
+    return absl::InvalidArgumentError(
+        "number_of_shards is out of the valid range");
+  }
+  if (parameters.required_fake_queries() < 0 ||
+      parameters.required_fake_queries() > kMaxRequiredFakeQueries) {
+    return absl::InvalidArgumentError(
+        "required_fake_queries is out of the valid range");
+  }
+  if (parameters.required_queries_per_shard() < 0 ||
+      parameters.required_queries_per_shard() > kMaxQueriesPerShard) {
+    return absl::InvalidArgumentError(
+        "required_queries_per_shard is out of the valid range");
+  }
+  if (std::numeric_limits<int>::max() / parameters.number_of_shards() <=
+      parameters.required_queries_per_shard()) {
+    return absl::InvalidArgumentError(
+        "The total number of required queries is too large");
+  }
+  if (parameters.required_queries_per_shard() != 0 &&
+      parameters.number_of_shards() * parameters.required_queries_per_shard() <=
+          parameters.required_fake_queries()) {
+    return absl::InvalidArgumentError(
+        "The number of required fake queries is too large");
+  }
+  return absl::OkStatus();
+}
+}  // namespace
+
+absl::Status ValidateParameters(const Parameters& parameters) {
+  if (auto status = ValidateCryptoParameters(parameters.crypto_parameters());
+      !status.ok()) {
+    return status;
+  }
+  return ValidateShardParameters(parameters.shard_parameters());
```

Comment:
Might be better to do

```
if (auto status = ValidateShardParameters(parameters.shard_parameters()); !status.ok()) {
  return status
}
return absl::OkStatus();
```

in case we add more parameter type.

Reply:
Done.